src/test/java/org/elasticsearch/rest/action/admin/indices/upgrade/UpgradeTest.java

```diff
@@ -142,32 +142,41 @@ public boolean apply(Object o) {
         logger.info("--> Nodes upgrade complete");
         logSegmentsState();
         
-        final HttpRequestBuilder httpClient = httpClient();
-
-        assertNotUpgraded(httpClient, null);
+        assertNotUpgraded(httpClient(), null);
         final String indexToUpgrade = "test" + randomInt(numIndexes - 1);
         
+        // #10213:
+        logger.info("--> Running upgrade on index " + indexToUpgrade + ", only ancient segments");
+        runUpgrade(httpClient(), indexToUpgrade, "only_ancient_segments", "true");
+
+        // TODO: why do we never test upgrading ancient segments here...?
+        // E.g. 0.90.x (Lucene 3.x) --> 1.x (Lucene 4.x)  ElasticsearchBackwardsCompatIntegrationTest.backwardsCompatibilityPath checks that
```

Comment:
> Hrm, 0.90.x was 4.x right?

Oh you're right!

OK I'll add a static index based on UpgradeReallyOldIndexTest on 1.x ..